src/config/api.rs

```diff
@@ -94,13 +96,23 @@ impl ApiConfig {
             .unwrap_or_default()
     }
 
+    pub fn get_timeout(&self) -> Option<Duration> {
```

Comment:
Small comment on the default behavior: I'd prefer to have the default method of the API config set it to 30s and use `None` for actual no timeout instead of zero.

I find less confusing to rely on the type system when possible instead of defining arbitrary behavior :slightly_smiling_face:

Reply:
Right, that makes sense. I'll try and get the changes in within the next day or so

Reply:
Thanks a ton!